src/State.mli

```diff
@@ -40,6 +40,8 @@ sig
 
   val run : init:state -> (unit -> 'a) -> 'a
   (** [run t] runs the thunk [t] which may perform state effects. *)
+
+  val register_printer : ?get:string -> ?set:(state -> string) -> unit -> unit
```

Comment:
I think `state -> string` should be `state -> string option`. Not sure about `get`...